### PR TITLE
UIE-204 Narrow Ajax Usage pt14

### PR DESCRIPTION
**Pull request by msilva-broad**
- shifting ajax usage to directly use ajax sub-areas for rest of the src/pages uses.
- improving mock types where possible.

### What
-

### Why
- improve code maintainability

### Testing strategy

- [ ]

### CODE REVIEW

File: src/pages/methods/workflow-details/WorkflowWrapper.test.tsx [hunk 1]
```diff
@@ -573,132 +563,6 @@ describe('workflows container', () => {
     expect(dialog2).not.toBeInTheDocument();
   });
 
-  it('renders edit method modal when corresponding button is pressed', async () => {
```
Comment (msilva-broad) [inline src/pages/methods/workflow-details/WorkflowWrapper.test.tsx]: oops, bad merge... hold on...

File: src/pages/workspaces/workspace/Workflows.js [hunk 2]
```diff
@@ -205,10 +208,7 @@ export const FindWorkflowModal = ({ namespace, name, ws, onDismiss }) => {
   // Lifecycle
   useOnMount(() => {
     const load = async () => {
-      const [featuredList, methods] = await Promise.all([
-        Ajax(signal).FirecloudBucket.getFeaturedMethods(),
-        Ajax(signal).Methods.list({ namespace: 'gatk' }),
-      ]);
+      const [featuredList, methods] = await Promise.all([FirecloudBucket(signal).getFeaturedMethods(), Methods(signal).list({ namespace: 'gatk' })]);
```
Comment (salonishah11) [inline src/pages/workspaces/workspace/Workflows.js:211]: Just a heads up, the `FindWorkflowModal` is getting replaced in PR #5184. So these Ajax calls in the modal would no longer be needed.

File: src/pages/workspaces/workspace/Workflows.test.js [hunk 3]
```diff
@@ -2,12 +2,17 @@ import { screen } from '@testing-library/react';
 import { act } from '@testing-library/react';
```
Comment (salonishah11) [inline src/pages/workspaces/workspace/Workflows.test.js:2]: This whole test file is also getting restructured in PR #5184

**Review by salonishah11 — APPROVED** [review]
Thank you for doing this!

File: src/pages/workspaces/workspace/Workflows.test.js (hunk 3 above)
Comment (msilva-broad) [reply src/pages/workspaces/workspace/Workflows.test.js:2]: no worries... assuming this one gets merged first, just take the ajax and types patterns forward :)...  add me as a reviewer on 5184 if you think it makes sense.

**Review by sam-schu — APPROVED** [review]
Thanks for the improvements!

**Pull request merged**